pytorch_lightning/callbacks/base.py

```diff
@@ -40,6 +40,11 @@ def state_identifier(self) -> str:
     def _legacy_state_identifier(self) -> Type:
         return type(self)
 
+    def _generate_state_identifier(self, **kwargs: Any) -> str:
+        attrs = ", ".join(f"{k}={v}" for k, v in kwargs.items())
```

Comment:
n00b question: do we need any checks that the key/values here are serializable as strings?

Reply:
I added the requirement in the doc string. Perhaps that's enough, considering this is only a helper method?
Would you have checked for ``__str__`` on the object?

Reply:
This function should use repr instead

```suggestion
        attrs = ", ".join(f"{repr(k)}={repr(v)}" for k, v in kwargs.items())
```